File: backend/LibraryApi.Domain/src/Entities/Author.cs
namespace LibraryApi.Domain.src.Entities;

public class Author : BaseEntity
{
  public string FirstName { get; set; }
  public string LastName { get; set; }
  public List<Book> Books { get; } = new List<Book>();
  public int YearOfBirth { get; set; }
}

File: frontend/src/types/Author.ts
import { Book } from "./Book"

export interface Author {
  id: string
  firstName: string
  lastName: string,
  yearOfBirth?: number
  books: Book[]
}

export type NewAuthor = Omit<Author, 'id' | 'books'>
export type ListAuthor = Omit<Author, 'books'>